tools/gulp/packaging/ts-compiler.ts

```diff
@@ -0,0 +1,43 @@
+import * as ts from 'typescript';
+import * as path from 'path';
+import * as fs from 'fs';
+import * as chalk from 'chalk';
+
+/** Reads a input file and transpiles it into a new file. */
+export function transpileFile(inputPath: string, outputPath: string, options: ts.CompilerOptions) {
+  const inputFile = fs.readFileSync(inputPath, 'utf-8');
+  const transpiled = ts.transpileModule(inputFile, { compilerOptions: options });
+
+  reportDiagnostics(transpiled.diagnostics);
+
+  fs.writeFileSync(outputPath, transpiled.outputText);
+
+  if (transpiled.sourceMapText) {
+    fs.writeFileSync(`${outputPath}.map`, transpiled.sourceMapText);
+  }
+}
+
+/** Formats the TypeScript diagnostics into a error string. */
+export function formatDiagnostics(diagnostics: ts.Diagnostic[], baseDir: string): string {
```

Comment:
Ugh this ends up in a lot of files. If we do this then we should have a folder `packaging/typescript`?